boards/native/Makefile.include

```diff
@@ -14,7 +14,23 @@ RESET ?= $(RIOTBOARD)/native/dist/reset.sh
 FLASHER ?=
 FLASHFILE ?= $(ELFFILE)
 
-TERMPROG ?= $(FLASHFILE)
+SETPYTERMFLAGS := 0
+ifdef RIOT_TERMINAL
+  SETPYTERMFLAGS = 0
+else ifdef TERMPROG
+  SETPYTERMFLAGS = 0
+else
+  SETPYTERMFLAGS = 1
+endif
+ifneq (,$(findstring pyterm,$(TERMPROG)))
+  SETPYTERMFLAGS = 1
+endif
+
+ifeq (1, $(SETPYTERMFLAGS))
+  TERMFLAGS += -n $(FLASHFILE)
+else
+  TERMPROG ?= $(FLASHFILE)
+endif
```

Comment:
That was my effort to keep application Makefiles as much as possible as they are - but that's overly complicated. I'm overhauling this.